models/test/mapper/TestPreviewMapper.php

```diff
@@ -0,0 +1,202 @@
+                $partId = $testPart->getIdentifier();
+                $navigationMode = $testPart->getNavigationMode();
+
+                if ($displaySubsectionTitle) {
+                    $section = $routeItem->getAssessmentSection();
+                } else {
+                    $sections = $routeItem->getAssessmentSections()->getArrayCopy();
+                    $section = $sections[0];
+                }
+
+                $sectionId = $section->getIdentifier();
+                $itemId = $itemRef->getIdentifier();
+
+                if ($lastSection != $sectionId) {
+                    $offsetSection = 0;
+                    $lastSection = $sectionId;
+                }
+
+                $itemUri = $itemRef->getHref();
+
+                $itemInfos = [
+                    'id' => $itemId,
+                    'uri' => $itemUri,
+                    'label' => $this->getItemLabel($itemUri),
+                    'position' => $offset,
+                    'occurrence' => $occurrence,
+                    'remainingAttempts' => 0,
+                    'answered' => 0,
+                    'flagged' => false,
+                    'viewed' => false,
+                    'categories' => [],
+                ];
+
+                if ($checkInformational) {
+                    $itemInfos['informational'] = $isItemInformational;
+                }
+
+                if ($itemRef->hasTimeLimits()) {
+                    $itemInfos['timeConstraint'] = null; //@TODO Implement as feature
+                }
+
+                if (!isset($map['parts'][$partId])) {
+                    $map['parts'][$partId]['id'] = $partId;
+                    $map['parts'][$partId]['label'] = $partId;
+                    $map['parts'][$partId]['position'] = $offset;
+                    $map['parts'][$partId]['isLinear'] = $navigationMode == NavigationMode::LINEAR;
+
+                    if ($testPart->hasTimeLimits()) {
+                        $map['parts'][$partId]['timeConstraint'] = null; //@TODO Implement as feature
+                    }
+                }
+
+                if (!isset($map['parts'][$partId]['sections'][$sectionId])) {
+                    $map['parts'][$partId]['sections'][$sectionId]['id'] = $sectionId;
+                    $map['parts'][$partId]['sections'][$sectionId]['label'] = $section->getTitle();
+                    $map['parts'][$partId]['sections'][$sectionId]['isCatAdaptive'] = false; //@TODO Implement as feature
+                    $map['parts'][$partId]['sections'][$sectionId]['position'] = $offset;
+
+                    if ($section->hasTimeLimits()) {
+                        $map['parts'][$partId]['sections'][$sectionId]['timeConstraint'] = null; //@TODO Implement as feature
+                    }
+                }
+
+                $map['parts'][$partId]['sections'][$sectionId]['items'][$itemId] = $itemInfos;
+
+                $this->updateStats($map, $itemInfos);
```

Comment:
maybe extracted into a update all stats , after mvp

Reply:
Yes, can be improved and we have test coverage for this :)